### PR TITLE
Optimizes particle holders

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

It isn't really an issue now, but these will be used more in future, so let's start off strong.

There's a lot of work going on here that doesn't really need to be happening, mostly off not knowing a trickTM.

Biggest one is vis_locs and vis_contents are linked lists, being in one requires being in another. Atoms clear out their vis_locs on Destroy, so we do not need to "own" references to things that have us in their vis_contents.

This combined with knowing our old loc's loc off Moved made the use of weakrefs here unneeded. Similarly, atoms inside atom movables qdel on when the upper layer is deleted, so most cases of the qdeleting signal were unneeded.

Also, we only cared about movement if we were an item (speaking of which, I swapped out the isitem stuff with a flag that gets passed into the new() call)

## Why It's Good For The Game

Speed

### CODE REVIEW

File: code/game/objects/effects/particle_holder.dm [hunk 1]
```diff
@@ -5,63 +5,52 @@
...
-	update_visual_contents(loc)
+	// /atom doesn't have vis_contents, /turf and /atom/movable do
+	var/atom/movable/lie_about_areas = loc
+	lie_about_areas.vis_contents += src
+	if(!ismovable(loc))
+		RegisterSignal(loc, COMSIG_PARENT_QDELETING, PROC_REF(immovable_deleted))
+
+	if(particle_flags & PARTICLE_ATTACH_MOB)
+		RegisterSignal(loc, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+	on_move(loc, null, NORTH)
```
Comment (san7890) [inline code/game/objects/effects/particle_holder.dm:29]: could be a nothing burger but i think this would be easier to "get" if args were named here

Comment (LemonInTheDark) [reply code/game/objects/effects/particle_holder.dm:29]: uh, sure

**Comment by Time-Green** [comment]
Merge conflict ;_;

**Comment by LemonInTheDark** [comment]
hhhhhhhh

**Pull request merged**